### PR TITLE
Week5 assignment

### PR DESCRIPTION
**Pull request by sanjaykashyap80**

### CODE REVIEW

File: Week-5/Exercise5.4/main.test.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+function add(a,b){
+    return (a+b);
+}
+test("adding two numbers", () => {
+    expect(add(2,2)).toBe(2+2)
```
Comment (ShrutikaSingh) [inline Week-5/Exercise5.4/main.test.js:5]: toBe(4)

think of more edge case, add(-1,0) toBe(-1)
add(-1.90932434,0) toBe(-1.90932434)

File: Week-5/Exercise5.4/main.test.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+})
+
+function diff(a,b){
+    return (a-b);
+}
+test("difference of two numbers", () => {
+    expect(diff(2,2)).toBe(2-2)
```
Comment (ShrutikaSingh) [inline Week-5/Exercise5.4/main.test.js:12]: toBe(0)

File: Week-5/exercise5.1.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      .then(resp => resp.json())
+      .then(users => {
+        return users;
+      });
+  }
+  const usersIt = fetchUsers();
+  usersIt.next().value.then(resp => console.log(resp));
+  
+  setTimeout(() => {
+    console.log("------------------------------------------------------")
```
Comment (ShrutikaSingh) [inline Week-5/exercise5.1.js:13]: ??

File: Week-5/exercise5.1.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+// Using generator function
+function* fetchUsers() {
+    yield fetch('https://jsonplaceholder.typicode.com/users')
```
Comment (ShrutikaSingh) [inline Week-5/exercise5.1.js:3]: store url is constants

File: Week-5/exercise5.1.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  const usersIt = fetchUsers();
+  usersIt.next().value.then(resp => console.log(resp));
+  
+  setTimeout(() => {
+    console.log("------------------------------------------------------") 
+  }, 0)
+
+  // Using async await
+  async function fetchUsers2() {
+    const response = await fetch('https://jsonplaceholder.typicode.com/users');
```
Comment (ShrutikaSingh) [inline Week-5/exercise5.1.js:18]: +1

File: Week-5/Exercise5.2.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+function isVowel(char) {
+    return "aeiou".includes(char);
+    }
+
+    function vowelCount(str) {
+        const vowelMap = new Map();
```
Comment (ShrutikaSingh) [inline Week-5/Exercise5.2.js:6]: looks good

File: Week-5/Exercise5.3.js [hunk 7]
```diff
@@ -0,0 +1,2 @@
+const hasDuplicate = arr => new Set(arr).size !== arr.length
```
Comment (ShrutikaSingh) [inline Week-5/Exercise5.3.js:1]: yup great that you have used sets, using set improves the performance
